### PR TITLE
Post oop done

### PR DESCRIPTION
**Pull request by nazmariam**
OOP exercise

[Demo](https://nazmariam/nazmariam.github.io/post-oop) |
[Code base](https://github.com/nazmariam/nazmariam.github.io/tree/master/post-oop)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@nazmariam Okay, there is some space for improvements yet.

Please, check yourself against the following list.

 * [ ] Implement a base class to inherit from
 * [ ] Employ default parameters
 * [ ] Each species is represented with its own class
 * [ ] No need to specify species at instantiation
 * [ ] Classes for species that do not have hands by natural design
       do not consequently have `hands` or any equivalent property
 * [ ] All inhabitants are stored in a container (array or object)
 * [ ] JS native features are intensively employed (`const`, `let`, `Array.map|join|forEach|...`, etc)
 * [ ] [OOP, SOLID and DRY](https://github.com/OleksiyRudenko/a-tiny-JS-world/blob/master/README.md#learn-on-your-own)
       principles are intensively employed
 
Optional level up (not required to implement):
 * [ ] Friends list is a list of objects refs rather than names (strings)
 * [ ] Cat-woman class is built employing composition rather than inheritance only
 
Bonus:
 * [ ] `toString` magic method; when implemented `print(inhabitant)` does the job as `.toString` is called implicitly
 * [ ] `this.constructor.name`; when used no need to store `species` property
 
[ES6 classes cheat-sheet](https://gist.github.com/OleksiyRudenko/672d39b08d9d0da4e179aca49876c58b)

Please, confirm all items above are positive. Please, ask a question if anything above requires clarification.

**Comment by nazmariam** [comment]
I checked, all items (except optional and bonus, which I want to do after 4th of January) are implemented I think.

File: submissions/nazmariam/post-oop/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// ======== OBJECTS DEFINITIONS ========
+class Citizen {
+    constructor(species, name, gender, say, legs, hands){
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.say = say;
+        this.legs = legs;
+        this.hands = hands;
+    }
```
Comment (OleksiyRudenko) [inline submissions/nazmariam/post-oop/index.js]: >  Classes for species that do not have hands by natural design
do not consequently have hands or any equivalent property

Once base class has hands prop, all derived classes do.

Please, fix.

File: submissions/nazmariam/post-oop/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    new Human('Jack White','male','A seven nation army couldn\'t hold me back')
+];
+// ======== OUTPUT ========
+citizens.forEach(el => {
+    print(`<div>
+        Hi! I'm a ${el.species}. My name is ${el.name}. My gender is: ${el.gender}
+        I have: ${el.legs} legs and ${el.hands} hands.
+        I speak like this: ${el.say}
+        </div>`);
+});
```
Comment (OleksiyRudenko) [inline submissions/nazmariam/post-oop/index.js]: Formation of a string for output must be a method of an appropriate class.

File: submissions/nazmariam/post-oop/index.js (hunk 1 above)
Comment (nazmariam) [reply submissions/nazmariam/post-oop/index.js]: Of course! Fixed it.

File: submissions/nazmariam/post-oop/index.js (hunk 2 above)
Comment (nazmariam) [reply submissions/nazmariam/post-oop/index.js]: Did a method sayAbout(). Please, take a look.

**Pull request merged**